### ISSUE TITLE
Is it possible to write my own loss function (criterion)using sklearn for torch, while the new loss function still use some original torch criterion?

### ISSUE THREAD

**Issue by brisker** [hughperkins/pytorch#26]
Is it possible to write my own loss function (criterion)using sklearn for torch, while the new loss function still use some original torch criterion?

**Comment by hughperkins** [comment]
I guess your question comes down to "can I pass a python function into lua, so that it becomes a lua function, in lua, that proxies into the python function?"

It seems like it should be possible to implement such a thing, no particular reason I can think of why that wouldnt be fairly straightforward to get working, but pretty sure that it's not implemented currently.  I dont have time to implement this at this time.  You would need to add such functionality yourself, into pytorch.

What works currently:
- passing an object from lua to python, such that the lua object is wrapped by a python object (or retrieving such an object out of the lua, from python)
- calling methods on this object, from python, which methods can return additional lua objects, which are wrapped into python objects, and returned to the python side

**Comment by brisker** [comment]
Thanks for your reply. Let me put my question more specifically:  the lua function produces several tensors, and I need to operate these tensors using numpy or sklearn , and return additional tensors , and after this, I need to operate these additional tensors back in lua. So is this fairly easy to perform using pytorch?

**Comment by brisker** [comment]
@hughperkins 
Thanks for your reply!
Let me make my question more specifically:
I want to write the code mainly in lua, and calling numpy and sklearn functions **in lua**  to operate on torch tensors. and the return tensors of numpy and sklearn needs to be back to lua again. 
Is that possible right now?

**Comment by hughperkins** [comment]
Yeah, so, if you call into the lua from python, then you can receive the tensors from lua into the python, eg by calling methods on the lua object which return them. Then you can process those using eg sklearn, then call back into lua, passing the new tensors as parameters.

On 9 December 2016 15:20:53 CET, numberjcc <notifications@github.com> wrote:
>@hughperkins 
>Thanks for your reply!
>Let me make my question more specifically:
>I want to write the code mainly in lua, and calling numpy and sklearn
>functions in lua to operate on torch tensors. and the return tensors of
>numpy and sklearn needs to be back to lua again. 
>Is that possible right now?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/hughperkins/pytorch/issues/26#issuecomment-266024526

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by brisker** [comment]
@hughperkins 
Thanks for your reply!
Even more specifically: how to call back into lua ?  It seems that there is no "require numpy"  or "require sklearn" in lua or torch.
note that I want to write the main code in lua.

**Comment by hughperkins** [comment]
Your 'main' should be in python. Python calls into lua, gets the lua tensor, then passes it to sklearn.

On 9 December 2016 16:27:32 CET, numberjcc <notifications@github.com> wrote:
>@hughperkins 
>Even more specifically: how to call back into lua ?  It seems that
>there is no "require numpy" . 
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/hughperkins/pytorch/issues/26#issuecomment-266040738

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by brisker** [comment]
@hughperkins 
Hi,I am using pytorch,  and after I finish training a model, I find the training epoch seems not enough, so I reload it and try to finetune(retrain), but why the training loss seems like the beginning training loss, just like I am training it from scratch?！ I got really confused about this bug! Help?
